### PR TITLE
fix: call indent lambda only when amount is positive

### PR DESCRIPTION
**Pull request by Amine-Mike**

### CODE REVIEW

File: lib/oppen/print_stack.rb [hunk 1]
```diff
@@ -246,7 +246,7 @@ def write(obj)
     #
     # @return [Nil]
     def indent(amount)
-      write genspace.call(amount)
+      write genspace.call(amount) if amount.positive?
```
Comment (stackmystack) [inline lib/oppen/print_stack.rb:251]: Going silent is a bad idea. Let's emit a warning if it's not. Check `warn`. Or should we throw?

File: test/customization_test.rb [hunk 2]
```diff
@@ -155,6 +155,17 @@ def customization_build_output(printer)
     printer = Oppen::Wadler.new(width: 45, space: ->(*_args) { '*' })
     _ { printer.output }.must_raise ArgumentError
   end
+
+  it 'must only call space when indent is positive' do
```
Comment (stackmystack) [inline test/customization_test.rb]: The phrasing is that of the old style.

File: test/customization_test.rb [hunk 3]
```diff
@@ -155,6 +155,17 @@ def customization_build_output(printer)
...
+    printer = Oppen::Wadler.new(width: 10, space: ->(_n) { '*******' })
+    printer.text 'Hello'
+    printer.break
+    printer.text 'World'
+    _(printer.output).must_equal <<~LANG.chomp
+      Hello
+      World
+    LANG
```
Comment (stackmystack) [inline test/customization_test.rb]: This doesn't seem to test positive indentation at a first glance. Could you explain how it does (in its doc comments).

**Review by stackmystack — CHANGES_REQUESTED** [review]

File: lib/oppen/print_stack.rb (hunk 1 above)
Comment (Amine-Mike) [reply lib/oppen/print_stack.rb:251]: No in this case we want to go silent. Indent is called after each new line, if indent is 0, it is not an error/warning, we just do not need indentation.

Comment (stackmystack) [reply lib/oppen/print_stack.rb:251]: But if it's negative … ?

Comment (Amine-Mike) [reply lib/oppen/print_stack.rb:251]: We will raise an error if it is negative.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## [unreleased]
 
+## v0.9.4 (25-10-2024)
+
+- Indent only when amount is positive.
+
```
Comment (stackmystack) [inline CHANGELOG.md]: No, this should go in unreleased. Only when we release we'd attribute a version and a date.

**Review by stackmystack — CHANGES_REQUESTED** [review]

**Review by stackmystack — APPROVED** [review]

**Pull request merged**